index.d.ts

```diff
@@ -100,29 +100,35 @@ interface Request {
   }
 }
 
+type KVValue<Value> = Promise<Value | null>
+
 declare module '@cloudflare/workers-types' {
   export interface KVNamespace {
-    get(key: string, type?: 'text'): Promise<string>
-    get(key: string, type?: 'json'): Promise<object>
-    get(key: string, type?: 'arrayBuffer'): Promise<ArrayBuffer>
-    get(key: string, type?: 'stream'): Promise<ReadableStream>
-    get(key: string, type?: 'text' | 'json' | 'arrayBuffer' | 'stream'): Promise<any>
+    get(key: string, type?: 'text'): KVValue<string>
+    get(key: string, type?: 'json'): KVValue<object>
```

Comment:
Does the type `json` actually get parsed into an object here? Or does it return a `string`?

Reply:
Parsed into an object/array/whatever else parseable by JSON.parse.

Reply:
Maybe this should be:

```ts
    get<T = unknown>(key: string, type?: 'json'): KVValue<T>
```

Reply:
Updated